streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceivedBlockTracker.scala

```diff
@@ -122,6 +161,26 @@ private[streaming] class ReceivedBlockTracker(
     }
   }
 
+  /**
+   * Allocate all unallocated blocks to the given batch.
+   * This event will get written to the write ahead log (if enabled).
+   */
+  def allocateBlocksToBatch(batchTime: Time): Unit = {
```

Comment:
Why have three versions of every func - e.g. allocateBlocksToBatch, allocateBlocksToBatchAync, and allocateBlocksToBatch0? Why not have a single version - allocateBlocksToBatch, which takes an optional parameter async: Boolean. It will call writeToLog with that parameter, and writeToLog can check that async parameter to decide what to do.